push.go

```diff
@@ -21,6 +21,21 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
+// cacheLookupReferenceFunc wraps a BlobCache into a
+// libimage.LookupReferenceFunc to allow for using a BlobCache during
+// image-copy operations.
+func cacheLookupReferenceFunc(directory string, compress types.LayerCompression) libimage.LookupReferenceFunc {
+	// NOTE: this prevents us from moving BlobCache around and generalizes
+	// the libimage API.
```

Comment:
```suggestion
	// Using a closure here allows us to reference a BlobCache without having to explicitly
	// maintain it in the libimage API.
```
… but even that feels more like a comment for the  `libimageOptions.{Source,Destination}LookupReferenceFunc` field definition (as a rationale for using that design), it’s not specific to this one use.